### PR TITLE
Pathway query

### PR DESCRIPTION
**Pull request by TimothyStiles**
@jordan-strasser has created a sweet SQL query to generate new metabolic pathways. This PR is a work in progress and @Koeng101 is going to refactor the SQL into distinct queries after Jordan writes and confirms that his test case in `pathways_test.go` works as intended.

### CODE REVIEW

**Review by TimothyStiles — COMMENTED** [review]
Did you mean to include the swap and db-shm and db-wal files? This seems like accidental cruft to me.

**Comment by jordan-strasser** [comment]
Didn’t intend to include that stuff, just didn’t know if it was safe to delete.

________________________________

**Comment by TimothyStiles** [comment]
@jordan-strasser tests aren't passing and it looks like the one you wrote fails by default. Could you get the test to work before @Koeng101 takes a look at it?

**Comment by TimothyStiles** [comment]
Also it's safe to delete the cruft. Just delete and push your changes 🚀

**Comment by jordan-strasser** [comment]
This is strange, because all the tests I wrote passed on my computer. Not sure why it’s failing here

________________________________

**Comment by Koeng101** [comment]
@jordan-strasser you can check why with "details"

For example:
```
pathways_test.go:11: Expected: len(result) = 9, Got:  0
--- FAIL: TestGetTotalPathways (0.00s)

```

Wonder why that is happening. Are the test files all there?

**Comment by jordan-strasser** [comment]
Oh the database isn’t there. Makes sense

________________________________

**Comment by Koeng101** [comment]
Uh oh. We didn't strip it down yet did we?

**Comment by Koeng101** [comment]
(make sure to make backups before this) you can probably just run a query to delete the uniprot rows not associated with the coli genome.

**Comment by jordan-strasser** [comment]
Is that necessary? I think I just didn’t properly commit the database

________________________________

**Comment by Koeng101** [comment]
Not necessary strictly speaking, but I'd be willing to bet Tim will require for the merge. If you get it working, I can do the row dropping.

**Comment by jordan-strasser** [comment]
Does this test db need to be smaller to merge? I wanted to use it to test hybrid organisms too once we add more genbank files

________________________________

**Comment by TimothyStiles** [comment]
There's more to be done review wise but for now @jordan-strasser could you fix error handling in `pathways.go`?

**Comment by TimothyStiles** [comment]
@jordan-strasser I just ran my code coverage tool and it looks like you also needs tests to check for error handling in 'pathways.go'.

**Comment by jordan-strasser** [comment]
Does it give a line number? I just checked the code and it looks like all errors are checked for.
________________________________

**Comment by TimothyStiles** [comment]
@jordan-strasser I did another round of review. Please make edits and ask as many questions as you need!

**Comment by TimothyStiles** [comment]
@jordan-strasser I did another round of review. Please make edits and ask as many questions as you need!

**Review by TimothyStiles — COMMENTED** [review]
I forgot to send some of my earlier comments for review.

**Comment by TimothyStiles** [comment]
@jordan-strasser the test database is above ~10mb and you forgot to stop tracking your .swp file. Could you make the test database smaller and delete the .swp file?

**Comment by jordan-strasser** [comment]
I don't see the .swp file anywhere, and it's not possible for me to get the db down to 10mb without destroying the GetTotalPathways function. Just dropped 2 extraneous tables and got it from 44 mb to 40 mb

File: pathways/pathways.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	if err != nil {
+		fmt.Println("Compound not available yet. Improving Alchemy soon.")
+		err.Error()
+		//whenever there's an error here we need to log desired compounds
+	}
+	return id
+}
+
+type pathdata struct {
+	Rxn_id, Prod_id, Sub_id, Lvl                          int
```
[the remark this answers is not in the document]
Comment (TimothyStiles) [reply pathways/pathways.go:56]: For example `Rxn_id` should be `ReactionId`. Also field names in Go structs are capitalized camel case with no underscores. @jordan-strasser

**Comment by TimothyStiles** [comment]
@jordan-strasser all tests and checks are passing but you still need to write tests to prove that error handling works correctly.

**Comment by jordan-strasser** [comment]
Do I need to specifically handle errors within the test functions? At each error in pathways.go I do log.Fatalf("...%d", err) to handle them internally so I can just have one thing returned

**Pull request merged**